lib/matplotlib/tests/test_backend_pdf.py

```diff
@@ -191,3 +193,22 @@ def psfont(*args, **kwargs):
     ax.text(0.5, 0.5, 'hello')
     with tempfile.TemporaryFile() as tmpfile, pytest.raises(ValueError):
         fig.savefig(tmpfile, format='pdf')
+
+
+def test_pdf_savefig_when_color_is_none():
+    backup_params = mpl.rcParams.copy()
+    mpl.rcParams.update(mpl.rcParamsDefault)
```

Comment:
@dstansby  The reason for using default rcParams is that in testing environment, `rcParams['_internal.classic_mode'] == True` and the resulting code path of this setting does not go through the error i.e. `gc._rgb` is always not `None`. In other words, this error (and the required fix) is very specific to the setting where `rcParams['_internal.classic_mode'] == False`, which happens to be the default behaviour. Admittedly, I over-killed the setting by set everything back to default rather than a single setting.